CRM/Export/BAO/ExportProcessor.php

```diff
@@ -428,6 +458,23 @@ public function getHeaderForRow($field) {
    * @return array
    */
   public function runQuery($params, $order, $returnProperties) {
+    $addressWhere = '';
+    $params = array_merge($params, $this->getWhereParams());
+    if ($this->isPostalableOnly) {
+      if (array_key_exists('street_address', $returnProperties)) {
+        $addressWhere = " civicrm_address.street_address <> ''";
+        if (array_key_exists('supplemental_address_1', $returnProperties)) {
+          // We need this to be an OR rather than AND on the street_address so, hack it in.
+          $addressOptions = CRM_Core_BAO_Setting::valueOptions(CRM_Core_BAO_Setting::SYSTEM_PREFERENCES_NAME,
+            'address_options', TRUE, NULL, TRUE
+          );
+          if (!empty($addressOptions['supplemental_address_1'])) {
+            $addressWhere .= " AND civicrm_address.supplemental_address_1 <> ''";
```

Comment:
@eileenmcnaughton I think this should be OR not AND!

Reply:
I think you are right - if street address is not empty or supplemental address is not empty then we have an address we can email - updated